packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -371,8 +377,18 @@ export async function scanParentDirs(
       'pnpm-lock.yaml',
       'bun.lock',
       'bun.lockb',
+      '.vercel/project.json',
     ],
   });
+
+  const vercelProjectJson = vercelProjectJsonPath
```

Comment:
This logic is not necessary. Builders have access to the project settings in the `config` object passed to the `build()` function.